ddtrace/internal/telemetry/writer.py

```diff
@@ -195,6 +201,28 @@ def app_started_event(self):
         }
         self.add_event(payload, "app-started")
 
+    def app_heartbeat_event(self):
+        # type: () -> None
+        if self._forked:
+            # TODO: Enable app-heartbeat on forks
+            #   Since we only send app-started events in the main process
+            #   any forked processes won't be able to access the list of
+            #   dependencies for this app, and therefore app-heartbeat won't
+            #   add much value today.
+            return
+
+        # DEV: Although the default flush interval is 60 seconds,
+        #   we want to explicitly control the flow here in case
+        #   the flush interval ever changes.
+        #
+        #   If the flush interval ever exceeds 60 seconds, then we
+        #   will queue a heartbeat event every flush:
+        #   60 <= X <= flush interval.
+        now = time.time()
+        if now - self._last_heartbeat >= TelemetryWriter.HEARTBEAT_INTERVAL:
```

Comment:
This is to reduce the total number of payloads sent? By not sending a separate one for the heartbeat every time? Or is there another purpose behind adding this behavior? Is that to reduce network traffic or are there other reasons?

Reply:
The reason for this is in case the flush interval is every not 60 seconds that we never send a heartbeat more than once every 60 seconds.

e.g. if flush interval was lowered to every 10 seconds, then we only want to send a heartbeat once every 6 flushes.

however, if the flush interval is every greater than 60 seconds, then the heartbeat flush interval will be once per-flush.